File: 2C++/Memory.cpp
class Memory : public Hardware{
    //atribut
    private:
        int frequency;
        int memorySize;
        bool supportsCuda;

    public:
        //konstruktor
        Memory(){
            this->frequency = 0; 
            this->memorySize = 0;
            this->supportsCuda = "";
        }
    
    //setter & getter frequency
    void setfrequency(int frequency){
        this->frequency = frequency;
    }
    int getfrequency(){
        return this->frequency;
    }

    //setter & getter memorySize
    void setmemorySize(int memorySize){
        this->memorySize = memorySize;
    }
    int getmemorySize(){
        return this->memorySize;
    }
    
    //setter & getter supportsCuda
    void setsupportsCuda(bool supportsCuda){
        this->supportsCuda = supportsCuda;
    }
    bool getsupportsCuda(){
        return this->supportsCuda;
    }

    ~Memory(){
    }
};

File: 1Java/Memory.java
public class Memory extends Hardware{
    //atribut
    private int frequency;
    private int memorySize;
    private boolean supportsCuda;
     
    //konstruktor kosong
    public Memory(){
    }

    //setter & getter frequency
    public void setfrequency(int frequency){
        this.frequency = frequency;
    }
    public int getfrequency(){
        return this.frequency;
    }

    //setter & getter memorySize
    public void setmemorySize(int memorySize){
        this.memorySize = memorySize;
    }
    public int getmemorySize(){
        return this.memorySize;
    }

    //setter & getter supportsCuda
    public void setsupportsCuda(boolean supportsCuda){
        this.supportsCuda = supportsCuda;
    }
    public boolean getsupportsCuda(){
        return this.supportsCuda;
    }
}
